src/change_trust_asset.js

```diff
@@ -0,0 +1,246 @@
+import clone from 'lodash/clone';
+import padEnd from 'lodash/padEnd';
+import trimEnd from 'lodash/trimEnd';
+import xdr from './generated/stellar-xdr_generated';
+import { Asset } from './asset';
+import { Keypair } from './keypair';
+import { LiquidityPoolFeeV18, getLiquidityPoolId } from './liquidity_pool_id';
+import { StrKey } from './strkey';
+
+/**
+ * ChangeTrustAsset class represents a trustline change to either a liquidity
+ * pool, the native asset (`XLM`) or an issued asset with an asset code / issuer
+ * account ID pair.
+ *
+ * In case of the native asset, the code will represent `XLM` while the issuer
+ * and liquidityPoolParameters will be empty. For an issued asset, the code and
+ * issuer will be valid and liquidityPoolParameters will be empty. For liquidity
+ * pools, the liquidityPoolParameters will be valid and the remaining two fields
+ * will be empty.
+ *
+ * @constructor
+ * @param {string} code - The asset code.
+ * @param {string} issuer - The account ID of the asset issuer.
+ * @param {LiquidityPoolParameters} liquidityPoolParameters – The liquidity pool parameters.
+ * @param {Asset} liquidityPoolParameters.asseta – The first asset in the Pool, it must respect the rule assetA < assetB.
+ * @param {Asset} liquidityPoolParameters.assetB – The second asset in the Pool, it must respect the rule assetA < assetB.
+ * @param {number} liquidityPoolParameters.fee – The liquidity pool fee. For now the only fee supported is `30`.
+ */
+export class ChangeTrustAsset {
+  constructor(code, issuer, liquidityPoolParameters) {
+    if (!code && !issuer && !liquidityPoolParameters) {
+      throw new Error(
+        'Must provide either code, issuer or liquidityPoolParameters'
+      );
+    }
+
+    // Validate code & issuer.
+    if ((code || issuer) && !/^[a-zA-Z0-9]{1,12}$/.test(code)) {
+      throw new Error(
+        'Asset code is invalid (maximum alphanumeric, 12 characters at max)'
+      );
+    }
+    if (code && String(code).toLowerCase() !== 'xlm' && !issuer) {
+      throw new Error('Issuer cannot be null');
+    }
+    if (issuer && !StrKey.isValidEd25519PublicKey(issuer)) {
+      throw new Error('Issuer is invalid');
+    }
+
+    this.code = code;
+    this.issuer = issuer;
+    if (code) {
+      // return early if it's an asset with code.
+      return;
+    }
+
+    // Validate liquidity pool params.
+    const { asseta, assetB, fee } = liquidityPoolParameters;
+    if (!asseta || !(asseta instanceof Asset)) {
```

Comment:
now that xdrgen is fixed, should these be assetA?

Reply:
I re-generated the variable names in https://github.com/stellar/js-stellar-base/pull/454/commits/714c66b12c4eae4f40dd874f650f75cd248206b9.